### ISSUE TITLE
extra line in fullchain

### ISSUE THREAD

**Issue by nosilver4u** [Neilpang/acme.sh#1940]
I'm wondering if it is intentional to have an extra line in between the certs in fullchain.cer?
The API for my CDN provider seems to throw a fit on that blank line, and if there isn't something that necessitate having it, would be nice to have it removed in a future release. I'm using version 2.8.0.

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Not a bug: fullchain.cer is the CA's PEM chain response verbatim; the blank line between certificates comes from the CA and is valid PEM (RFC 7468), so the CDN parser should accept it.

**Comment by nosilver4u** [comment]
I no longer see line-breaks in fullchain.cer. It's possible I worked around it somehow, but I can't recall at this point. That's fair enough though about the cert being direct from the CA and fully valid. If I have any trouble with it in the future, I'll work with our vendors to be sure they can handle line breaks properly :)

*Just figured out where I worked around it, and it's working fine for us. In case anyone else has issues, I'm running a post-generation script that parses out empty lines with this:
`sed -i '/^$/d' $DOMAIN_CERT`

$DOMAIN_CERT is a bash variable pointing at fullchain.cer.

**Comment by neilpang** [comment]
Hi  @nosilver4u  ,

I just checked again on one of my server.  it seems there is no extra empty line in the fullchain.cer between the certificates.

Can you please confirm it again?  I remember we fixed it some time ago.


Thanks
-Neil

**Comment by nosilver4u** [comment]
Sure, I just ran it again without my extra script and can see that fullchain.cer has a blank line between each of the certs, matching what is in ca.cer.

**Comment by neilpang** [comment]
@nosilver4u  sorry,  I used zerossl,  it returns fullchain without any empty lines.

but I just fix the code to remove any empty lines.  letsencrypt returns cert with empty lines.

**Comment by nosilver4u** [comment]
Ah, that makes sense, I'll check it out. Can I just do the built-in update, or do I need to do anything different to apply the fix to my local install of acme.sh?

**Comment by nosilver4u** [comment]
Oh, I see you committed it to 'dev', I'll test that out.

**Comment by nosilver4u** [comment]
Yup, it works. Both fullchain.cer and ca.cer have no more empty lines.

**Comment by neilpang** [comment]
it will be auto upgraded when merged to master.
